src/mlpack/methods/ann/layer/add_merge_impl.hpp

```diff
@@ -41,6 +41,17 @@ AddMerge<InputDataType, OutputDataType, CustomLayers...>::AddMerge(
   // Nothing to do here.
 }
 
+template<typename InputDataType, typename OutputDataType,
+         typename... CustomLayers>
+AddMerge<InputDataType, OutputDataType, CustomLayers...>::AddMerge(
+    const AddMerge& network) :
+    model(network.model),
+    run(network.run),
+    ownsLayers(network.ownsLayers)
```

Comment:
I will add this in next commit.

Reply:
Hey, @rcurtin I think we don't need to directly copy `network.network` because it will copy the addresses to previous layer and arma::mat of weight, gradient etc. and we will face the same problem again we are to solve.  I think we need to add layer by using `copyVisitor` and create a new `network`. I have tried it. What do you suggest ?

Reply:
Right, exactly, I think we'll need to use a `copyVisitor` for that. :+1:

Reply:
Hey @rcurtin I tried to copy the network by using copyvisitor but it is causing error.

Reply:
Can you describe what went wrong?  "it is causing error" isn't descriptive enough for me to be able to help. :)

Reply:
@rcurtin The error is 
```
error: addition: incompatible matrix dimensions: 0x0 and 7x1
terminate called after throwing an instance of 'std::logic_error'
  what():  addition: incompatible matrix dimensions: 0x0 and 7x1
```
Sorry for the previous lame response at that time the error was not quite clear. I will try to find out what can be the reason.

Reply:
I see, maybe this will only work correctly once all the copy constructors of other layers are working correctly.  So let's leave this comment for later I guess.